### PR TITLE
Add JReleaser

### PR DESCRIPTION
**Pull request by aalmiray**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

**Permissions**: JReleaser is a tool that creates releases for any kind of project. As such it must be able to read and write files from any directory where the user has placed their sources. For this reason `--filesystem=host` was added to the set of permissions.

**Requirements**: JReleaser requires the `org.freedesktop.Sdk.Extension.openjdk` extension.

### CODE REVIEW

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by aalmiray** [comment]
"Build" step completed successfully but the "Check for AppStream.xml" step failed with

```
stat: cannot stat 'builddir/*/share/app-info/xmls/org.jreleaser.cli.xml.gz': No such file or directory
```

Any hints on how to fix this would be appreciated 😅

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

File: org.jreleaser.cli.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk
+appstream-compose: false
+command: jreleaser 
+finish-args:
+  - --env=PATH=/app/jre/bin:/usr/bin:/app/bin
+  - --env=JAVA_HOME=/app/jre
+  - --share=network
+  - --filesystem=host
```
Comment (hfiguiere) [inline org.jreleaser.cli.yml:12]: why access to the whole host?

File: org.jreleaser.cli.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk/install.sh
+
+  - name: jreleaser
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 bin/jreleaser /app/bin/jreleaser
```
Comment (hfiguiere) [inline org.jreleaser.cli.yml:22]: does this use a vulnerable log4j?

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by aalmiray** [comment]
Works as expected 🎉 

```
$ flatpak run org.jreleaser.cli release -y
[INFO]  JReleaser 0.9.1
[INFO]  Configuring with jreleaser.yml
[INFO]    - basedir set to /home/ubuntu/app
[INFO]  Reading configuration
[INFO]  Loading variables from /home/ubuntu/.jreleaser/config.toml
[INFO]  Validating configuration
[INFO]  Project version set to 1.0.0
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2021-12-18T15:20:03.130182305Z
[INFO]  HEAD is at 4df324c
[INFO]  Platform is linux-x86_64
[INFO]  dryrun set to true
[INFO]  Generating changelog: out/jreleaser/release/CHANGELOG.md
[INFO]  Calculating checksums
[INFO]    [checksum] target/distributions/app/app-1.0.0.zip.sha256
[INFO]    [checksum] target/distributions/app/app-1.0.0.tar.gz.sha256
[INFO]  Signing files
[INFO]  Signing is not enabled. Skipping
[INFO]  Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/aalmiray/app
[INFO]   - uploading app-1.0.0.zip
[INFO]   - uploading app-1.0.0.tar.gz
[INFO]   - uploading checksums_sha256.txt
[INFO]  Writing output properties to out/jreleaser/output.properties
[INFO]  JReleaser succeeded after 7.188 s
```

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

File: org.jreleaser.cli.yml (hunk 2 above)
Comment (aalmiray) [reply org.jreleaser.cli.yml:22]: No. If you look at the manifest file you'll see every single JAR file required by this tool. Log4j is not explicitly listed nor it's found in a repackaged form.

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

File: org.jreleaser.cli.yml (hunk 1 above)
Comment (aalmiray) [reply org.jreleaser.cli.yml:12]: As stated above: "Permissions: JReleaser is a tool that creates releases for any kind of project. As such it must be able to read and write files from any directory where the user has placed their sources. For this reason --filesystem=host was added to the set of permissions."

The tool can't know for sure which directories a user would like to use for hosting their code. Should it perhaps be restricted to just "home" ?

**Comment by aalmiray** [comment]
bot, build org.jreleaser.cli

**Comment by nedrichards** [comment]
it's not a requirement for a console app but given that you do have a cool looking jetpack duke icon it'd be nice to see an icon in future so you could benefit more from discoverability updates to the website and other software stores.

**Comment by aalmiray** [comment]
@nedrichards I added an icon to the manifest in the last commit. Could you tell me where else should I put it so that it becomes visible? Thank you for merging! :smile:

**Pull request closed without merging**